### ISSUE TITLE
Gradient clipping and monitoring

### ISSUE THREAD

**Issue by jan-forest** [jan-forest/autoencodix_package#193]
In the old Autoencodix framework gradient clipping was hardcoded with a high threshold: https://github.com/jan-forest/autoencodix/blob/c5933368504224562c48017c654ac127f2b2261d/src/utils/utils.py#L939 
In the new `_general_trainer.py` there is no gradient clipping added. However, I experience some instability during the training of large Ontix which might be related to exploding gradients. Hence, we should consider implementing:
- Option to monitor and warn when gradients explode
- Option to adjust gradient clipping threshold

Check also if vanishing gradients can also be monitored and warned as well.

**Comment by jan-forest** [comment]
Added gradient clipping in current dev commit. Will output a warning of clipping and reports total norm compared to max norm. 
Max norm is adjustable via config.
